src/context.ts

```diff
@@ -0,0 +1,28 @@
+import { prisma, Prisma, User } from "./generated/prisma-client";
+import Cookies from "universal-cookie";
+import { resolveUserUsingJWT } from "./utils/resolveUser";
+
+export type AppContext = {
+  prisma: Prisma;
+  token: string;
+  auth: {
+    user: User;
+    isAuthenticated: boolean;
+  };
+};
+
+export const context = async ({ req }): Promise<AppContext> => {
+  const cookies = new Cookies(req && req.headers && req.headers.cookie);
```

Comment:
Is it possible for `new Cookies(req && req.headers && req.headers.cookie)` to fail with `req = null` or `req.headers = null` cases? Find it a little hard to determine if `new Cookies(null)` is a good idea

Reply:
It will certainly not. The tests in fact run this context object. It just falls back to the fallback token.